cli/config/configurator.go

```diff
@@ -55,11 +55,19 @@ func (c Configurator) WithErrorHandler(fn errorHandlerFn) Configurator {
 }
 
 func (c Configurator) Start(ctx context.Context, prev *Config, flags agentConfig.Flags) error {
-	c.flags = flags
-	serverURL, err := c.getServerURL(prev, flags)
-	c.finalServerURL = serverURL
-	if err != nil {
-		return err
+	c.flags = &flags
```

Comment:
the original `flags` value was passed around in functions call, on top of being saved to the instance. this was confusing to follow and created 2 different copies of the same value, so modifications were almost impossible to track